ibis/backends/mssql/compiler.py

```diff
@@ -457,3 +457,25 @@ def visit_All(self, op, *, arg, where):
         if where is not None:
             arg = self.if_(where, arg, NULL)
         return sge.Min(this=arg)
+
+    def visit_Select(self, op, *, parent, selections, predicates, sort_keys):
+        # if we've constructed a useless projection return the parent relation
+        if not selections and not predicates and not sort_keys:
+            return parent
+
+        result = parent
+
+        if selections:
+            result = sg.select(*self._cleanup_names(selections), copy=False).from_(
+                result, copy=False
+            )
+
+        if predicates:
+            # TODO: copy=False should be allowed here, but sqlglot has a bug
+            # that prevents it from generating the correct code when copy=False
+            result = result.where(*predicates, copy=True)
```

Comment:
@kszucs I override `visit_Select` here to workaround a sqlglot bug. I haven't yet been able to make a simple reprex.

Reply:
It is possibly not a bug in SQLGlot, but that we mutate `parent` here. I think we need to check whether parent is a sqlglot select instance first, and if it is, then explicitly copy it instead of just assigning at line 466.

Reply:
That doesn't seem to work, so I'll keep this code as is.